tensorboard/uploader/uploader_main.py

```diff
@@ -541,11 +678,31 @@ def _get_intent(flags):
         raise base_plugin.FlagsError("Must specify subcommand (try --help).")
     if cmd == _SUBCOMMAND_KEY_UPLOAD:
         if flags.logdir:
-            return _UploadIntent(os.path.expanduser(flags.logdir))
+            return _UploadIntent(
+                os.path.expanduser(flags.logdir),
+                name=flags.name,
+                description=flags.description,
+            )
         else:
             raise base_plugin.FlagsError(
                 "Must specify directory to upload via `--logdir`."
             )
+    if cmd == _SUBCOMMAND_KEY_UPDATE_METADATA:
+        if flags.experiment_id:
+            if flags.name is not None or flags.description is not None:
+                return _UpdateMetadataIntent(
+                    flags.experiment_id,
+                    name=flags.name,
+                    description=flags.description,
+                )
+            else:
+                raise base_plugin.FlagsError(
+                    "Must specify either `--name` or `--description`."
+                )
+        else:
+            raise base_plugin.FlagsError(
+                "Must specify experiment to modify via `--experiment_id`."
+            )
```

Comment:
nit: Can we use early returns here to avoid pushing the code more and
more to the right?

```python
        if not flags.experiment_id:
            raise base_plugin.FlagsError(
                "Must specify experiment to modify via `--experiment_id`."
            )
        if flags.name is None and flags.description is None:
            raise base_plugin.FlagsError(
                "Must specify either `--name` or `--description`."
            )
        return _UpdateMetadataIntent(
            flags.experiment_id,
            name=flags.name,
            description=flags.description,
        )
```